app/src/main/kotlin/com/wire/android/ui/WireActivity.kt

```diff
@@ -263,6 +263,35 @@ class WireActivity : AppCompatActivity() {
                 )
             }
 
+            if (shouldShowTeamAppLockDialog) {
+                TeamAppLockFeatureFlagDialog(
+                    isTeamAppLockEnabled = isTeamAppLockEnabled,
+                    onConfirm = {
+                        featureFlagNotificationViewModel.dismissTeamAppLockDialog()
+                        if (isTeamAppLockEnabled) {
+                            val isUserAppLockSet =
+                                featureFlagNotificationViewModel.isUserAppLockSet()
+                            // No need to setup another app lock if the user already has one
+                            if (!isUserAppLockSet) {
+                                Intent(this@WireActivity, AppLockActivity::class.java)
+                                    .apply {
+                                        putExtra(AppLockActivity.SET_TEAM_APP_LOCK, true)
+                                    }.also {
+                                        startActivity(it)
+                                    }
+                            } else {
+                                featureFlagNotificationViewModel.markTeamAppLockStatusAsNot()
+                            }
+                        } else {
+                            with(featureFlagNotificationViewModel) {
+                                markTeamAppLockStatusAsNot()
+                                clearTeamAppLockPasscode()
```

Comment:
question: what will happen if the user first set the applock manually and then the team setting is changed to enforce applock? Then user will have to set a new passcode? 
Similar question when the team setting is disabled after some time - will then app be still locked because the app still keeps the passcode that user manually created before team setting enforced it?

Wouldn't be better then to just have a single field to keep the passcode and another field to determine which type it is?

Reply:
> what will happen if the user first set the applock manually and then the team setting is changed to enforce applock? Then user will have to set a new passcode?

No just he will be notified by a dialog that the team setting has been changed.

>  when the team setting is disabled after some time - will then app be still locked because the app still keeps the passcode that user manually created before team setting enforced it?

Yes will stay locked, once unlocked the user will see a dialog informing about the team setting change, and then passcode will be removed

Wouldn't be better then to just have a single field to keep the passcode and another field to determine which type it is?

I thought about this way, but since we are clearing the passcode when disabling it, I just decided to distinguish between those and go simpler